ui/pages/confirmations/components/confirm/row/dataTree.tsx

```diff
@@ -72,18 +156,28 @@ const DataField = memo(
         <DataTree
           data={value}
           isPermit={isPermit}
+          primaryType={primaryType}
           tokenDecimals={tokenDecimals}
         />
       );
     }
-    if (isPermit && label === 'value') {
+
```

Comment:
These small functions `isDate`, `isTokenUnits` can be extracted outside the component.

Not required, but I think it is also possible to create a Map of `type -> renderer`.

Reply:
moved logic into small functions 👍🏼 

will keep the rest as is for now